pkg/helpers/kubeconfig/smart_merge.go

```diff
@@ -47,6 +47,7 @@ func CreateConfig(namespace, userName string, clientCfg *restclient.Config) (*cl
 	if err != nil {
 		return nil, err
 	}
+	userName = userName + "/" + clusterNick
```

Comment:
You'd need to move this after `getContextNicknameFromConfig` otherwise you'll end up with context having `clusterNick` twice.
I also wonder if not reversing the order, even though this seems intuitive, contextNick has them reversed so it'll be more reasonable to keep that same approach here as well.